miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java

```diff
@@ -567,9 +570,7 @@ public ModelAndView setupFormByDilution(@PathVariable Long dilutionId, ModelMap
 
   private ModelAndView setupForm(User user, Library library, ModelMap model) throws IOException {
 
-    if (library == null) {
-      throw new SecurityException("No such Library.");
-    }
+    if (library == null) throw new NotFoundException("No library found");
 
     model.put("formObj", library);
     model.put("library", library);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "library" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)